File: XamarinStore.iOS/Views/ProductDescriptionView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using MonoTouch.UIKit;

namespace XamarinStore
{
	public class ProductDescriptionView : UIView
	{
		public UILabel Name { get; set; }
		public UILabel DescriptionLabel { get; set; }
		public UILabel Price { get; set; }
		const float padding = 20f;
		const float priceWidth = 50f;
		public ProductDescriptionView(Product product) : this()
		{
			this.Update(product);
		}
		public ProductDescriptionView ()
		{
			//this.TranslatesAutoresizingMaskIntoConstraints = false;
			Name = new UILabel
				{
					Text = "Name",
					BackgroundColor = UIColor.Clear,
					TextColor = UIColor.DarkGray,
					Font = UIFont.SystemFontOfSize(25f),
					TranslatesAutoresizingMaskIntoConstraints = false,
				};
			Name.SizeToFit();
			this.AddSubview(Name);
			

			DescriptionLabel = new UILabel
				{
					BackgroundColor = UIColor.Clear,
					TextColor = UIColor.DarkGray,
					TranslatesAutoresizingMaskIntoConstraints = false,
					Font = UIFont.SystemFontOfSize(12),
					LineBreakMode = UILineBreakMode.WordWrap,
					Lines = 0
				};
			this.AddSubview(DescriptionLabel);

			Price = new UILabel
				{
					BackgroundColor = UIColor.Clear,
					Text = "Price",
					TextColor = Color.Blue,
					TranslatesAutoresizingMaskIntoConstraints = false,
				};
			Price.SizeToFit();
			this.AddSubview(Price);
		}
		public void Update(Product product)
		{
			Name.Text = product.Name;
			DescriptionLabel.Text = product.Description;
			Price.Text = product.PriceDescription;
		}
		public override void LayoutSubviews()
		{
			base.LayoutSubviews();
			var bounds = Bounds;
			var frame = Name.Frame;

			frame.Width = bounds.Width - (priceWidth + padding *2);
			frame.Y = frame.X = padding;
			Name.Frame = frame;

			frame = Price.Frame;
			frame.Y = padding + (Name.Frame.Height - frame.Height)/2;
			frame.X = Name.Frame.Right + padding;
			frame.Width = priceWidth;
			Price.Frame = frame;

			frame = bounds;
			frame.Y = Name.Frame.Bottom;
			frame.X = padding;
			frame.Width -= padding*2;
			frame.Height -= frame.Y ;
			DescriptionLabel.Frame = frame;
		}
	}
}

File: F#/XamarinStore.iOS/Views/ProductDescriptionView.fs
namespace XamarinStore

open System
open System.Collections.Generic
open System.Linq
open System.Text
open MonoTouch.UIKit
open Product

type ProductDescriptionView(product:Product) as this =
    inherit UIView()

    let padding = 20.0f
    let priceWidth = 50.0f

    let name = new UILabel( Text = "Name",
                                   BackgroundColor = UIColor.Clear,
                                   TextColor = UIColor.DarkGray,
                                   Font = UIFont.SystemFontOfSize(25.0f),
                                   TranslatesAutoresizingMaskIntoConstraints = false)

    let descriptionLabel = new UILabel( BackgroundColor = UIColor.Clear,
                                        TextColor = UIColor.DarkGray,
                                        TranslatesAutoresizingMaskIntoConstraints = false,
                                        Font = UIFont.SystemFontOfSize(12.0f),
                                        LineBreakMode = UILineBreakMode.WordWrap,
                                        Lines = 0)
    let price = new UILabel( BackgroundColor = UIColor.Clear,
                             Text = "Price",
                             TextColor = Color.Blue.ToUIColor(),
                             TranslatesAutoresizingMaskIntoConstraints = false)

    do name.SizeToFit()
       this.AddSubview(name)

       this.AddSubview(descriptionLabel)

       price.SizeToFit()
       this.AddSubview(price)

       name.Text <- product.Name
       descriptionLabel.Text <- product.Description
       price.Text <- product.GetPriceDescription()
    
    member val Name = name with get,set
    member val DescriptionLabel = descriptionLabel with get,set
    member val Price = price with get,set

    override this.LayoutSubviews() =
        base.LayoutSubviews()
        let bounds = this.Bounds
        let mutable frame = this.Name.Frame

        frame.Width <- bounds.Width - (priceWidth + padding * 2.0f)
        frame.Y <- padding
        frame.X <- padding
        this.Name.Frame <- frame

        frame <- this.Price.Frame
        frame.Y <- padding + (this.Name.Frame.Height - frame.Height)/2.0f
        frame.X <- this.Name.Frame.Right + padding
        frame.Width <- priceWidth
        this.Price.Frame <- frame

        let mutable frame = bounds
        frame.Y <- this.Name.Frame.Bottom
        frame.X <- padding
        frame.Width <- frame.Width - padding*2.0f
        frame.Height <- frame.Height - frame.Y
        this.DescriptionLabel.Frame <- frame
